src/server/lib/okta/tokens.ts

```diff
@@ -0,0 +1,167 @@
+import { Request, CookieOptions } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import OktaJwtVerifier from '@okta/jwt-verifier';
+import { joinUrl } from '@guardian/libs';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { logger } from '@/server/lib/serverSideLogger';
+import { GU_DOMAIN } from '@/server/models/Configuration';
+
+/**
+ * Configuration
+ */
+const { okta, baseUri, stage } = getConfiguration();
+
+/**
+ * OAuth Token Name Type
+ */
+type OAuthCookieNames =
+  | typeof OAuthAccessTokenCookieName
+  | typeof OAuthIdTokenCookieName;
+
+/**
+ * Name of the OAuth Access Token cookie to set
+ * @type {string}
+ */
+const OAuthAccessTokenCookieName = 'GU_ACCESS_TOKEN';
+
+/**
+ * Name of the OAuth ID Token cookie to set
+ * @type {string}
+ */
+const OAuthIdTokenCookieName = 'GU_ID_TOKEN';
+
+const issuerDomain = () => {
+  // if we're in cypress mocked, we need to add https to the issuer domain, just to get it
+  // to validate, even if we never actually use the verifier in cypress mocked tests
+  if (process.env.RUNNING_IN_CYPRESS_MOCKED === 'true') {
+    return okta.orgUrl.replace('http://', 'https://');
+  }
+  // If we're in DEV, but we're developing against okta CODE environment, the issuer domain
+  // should be the okta CODE domain, hence we need this check to get the issuer right
+  if (stage === 'DEV' && process.env.GU_OKTA_ENV_COOKIE === 'CODE') {
+    return `https://profile.${GU_DOMAIN.CODE}/`;
+  }
+  return okta.orgUrl;
+};
+
+/**
+ * @name oauthTokenVerifier
+ * @type {OktaJwtVerifier}
+ * @description Okta JWT Verifier
+ */
+const oauthTokenVerifier = new OktaJwtVerifier({
+  issuer: joinUrl(issuerDomain(), '/oauth2/', okta.authServerId),
+});
+
+export const verifyAccessToken = async (token: string) => {
+  try {
+    const jwt = await oauthTokenVerifier.verifyAccessToken(
+      token,
+      issuerDomain(),
+    );
+    return jwt;
+  } catch (error) {
+    logger.error('Okta | Access Token | Verification Error', error);
+  }
+};
+
+export const verifyIdToken = async (token: string) => {
+  try {
+    const jwt = await oauthTokenVerifier.verifyIdToken(token, okta.clientId);
+    return jwt;
+  } catch (error) {
+    logger.error('Okta | ID Token | Verification Error', error);
+  }
+};
+
+/**
+ * Token cookie options
+ * Expiry is set to 1 hour by default, which matches the token expiry
+ */
+const OAuthTokenCookieOptions = (): CookieOptions => ({
+  httpOnly: true,
+  secure: !baseUri.includes('localhost'),
+  signed: !baseUri.includes('localhost'),
+  sameSite: 'lax',
+  maxAge: 3600000,
```

Comment:
There's no reason why we shouldn't move these numbers into constants or types somewhere central. Seems like a more reliable way to ensure we get the right times!